### ISSUE TITLE
Maintaining a Session and req.session.authenticated undefined

### ISSUE THREAD

**Issue by dottodot** [tjwebb/sails-auth#28]
I can can login using passport-local and the session is created in my mongodb, but for any subsequent requests req.session.passport is always empty.

Also req.session.authenticated in the sessionAuth policy is always undefined.

Not sure if I'm missing something or if it's a bug, I've setup others sails projects with passport and not had this issue.

**Comment by dottodot** [comment]
Additionally req.user is always undefined too, so I'm obviously missing pretty important.

**Comment by tjwebb** [comment]
Can you post your config/policies.js file?

**Comment by dottodot** [comment]
```
module.exports.policies = {

  '*': [
    'basicAuth',
    'passport',
    'sessionAuth',
    'ModelPolicy',
    'AuditPolicy',
    'OwnerPolicy',
    'PermissionPolicy',
    'RolePolicy',
    'localize'
  ],

  AuthController: {
    '*': [ 'passport' ]
  },

  LanguageController: {
    '*': true
  }

};
```

**Comment by tjwebb** [comment]
Which controller is req.user undefined inside of?

**Comment by yohanesyuen** [comment]
I am currently exploring authentication options for use in my sails app.

The documentation suggests to use the following in ```config/policies.js```:

```
'*': [ 'basicAuth', 'passport', 'sessionAuth' ],

  AuthController: {
    '*': [ 'passport' ]
  }
```

`sessionAuth` tests for req.session.authenticated before allowing a user through but `req.session.authenticated` is only set in `basicAuth` but from what I observe, it is never set in `passport`

**Comment by dottodot** [comment]
OK the first place where req.user is undefined is in api/polices/passport.js and in api/services/passport.js

I haven't worked out where req.user is set to trace why isn't not working.

Because it's alway undefined it means I can't use sails-permissions as AuditPolicy.js uses req.user.id

**Comment by bgulotta** [comment]
i'm getting the same behavior. i can authenticate using passport-local posting to auth/local, but subsequent requests are failing. i've added some debug logging and sails.sid is getting sent on subsequent requests. so the issue isn't there. i'll do some more digging and see what i can find.

**Comment by dottodot** [comment]
OK digging a bit deeper and it looks like the only place that req.user and req.session.authenticated are set is in basicAuth, so as I'm not using basicAuth this is why it's not working.

What I can't work out though is why req.session is always empty on subsequent requests.

**Comment by dottodot** [comment]
I've found the following seems to fix my issues.

In services/passport.js change
```
passport.serializeUser(function (user, next) {
  next(null, user);
});
```
to
```
passport.serializeUser(function (user, next) {
  next(null, user.id);
});
```
then in AuthContoller add ```req.session.authenticated = true;``` on successful login

**Comment by bgulotta** [comment]
@dottodot In my fork, I have added 2 new unit tests for this scenario. They were failing, I updated the code with your suggested changes and they are passing now.

**Comment by tjwebb** [comment]
is this resolved also via #30?

**Comment by bgulotta** [comment]
That's correct. This can be closed
